src/Codeworx.Identity.Primitives/UriBuilder.cs

```diff
@@ -15,21 +15,28 @@ public class UriBuilder
         private ImmutableArray<string> _segments;
 
         public UriBuilder(string baseUrl)
-            : this(new Uri(baseUrl))
+            : this(new Uri(baseUrl, UriKind.RelativeOrAbsolute))
```

Comment:
This should be solved in a different way. The UriBuilder should not be able to handle relative urls. Instead the calling side should detect relative Uris and combine the Servers BaseUri with the relative part before providing it to the UriBuilder Constructor

Reply:
Moved handling of relative URIs to the calling service.